homepage/homepage/app/docs/[framework]/[...slug]/design-patterns/form.mdx

```diff
@@ -0,0 +1,369 @@
+import { CodeGroup } from "@/components/forMdx";
+
+# Creating and updating CoValues in a form
+
+Normally we implement forms using
+[just the onSubmit handler](https://react.dev/reference/react-dom/components/form#handle-form-submission-on-the-client),
+or by making [a controlled form with useState](https://christinakozanian.medium.com/building-controlled-forms-with-usestate-in-react-f9053ad255a0),
+or by using special libraries like [react-hook-form](https://www.react-hook-form.com).
+
+
+In Jazz, we can do something simpler and more powerful, because CoValues give us reactive,
+persisted state which we can use to directly edit live objects, as well as to represent auto-saved drafts.
+
+## Updating a CoValue
+
+You simply assign the new value directly as changes happen. These changes are synced to the server, which means
+you don't need to handle form submissions either.
+
+<CodeGroup>
+```tsx
+<input
+  type="text"
+  value={order.name}
+  onChange={(e) => order.name = e.target.value}
+/>
+```
+</CodeGroup>
+
+This means we can write update forms in fewer lines of code.
+
+## Creating a CoValue
+
+However, when creating a CoValue, the CoValue does not exist yet, so we don't have the advantages previously mentioned.
+
+There's a way around this, and it provides unexpected benefits too.
+
+### Using a Draft CoValue
+
+Let's say we have a CoValue called `BubbleTeaOrder`. We can create a "draft" CoValue,
+which is an empty version of a `BubbleTeaOrder`, that we can then modify when we are "creating"
+a new CoValue.
+
+A `DraftBubbleTeaOrder` is essentially a copy of `BubbleTeaOrder`, but with all the fields made optional.
+
+<CodeGroup>
+```tsx
+// schema.ts
+export class BubbleTeaOrder extends CoMap {
+  name = co.string;
+}
+
+export class DraftBubbleTeaOrder extends CoMap {
+  name = co.optional.string;
+}
+```
+</CodeGroup>
+
+Then, each Account will have a `DraftBubbleTeaOrder`.
+
+<CodeGroup>
+```tsx
+// schema.ts
+export class AccountRoot extends CoMap {
+  draft = co.ref(DraftBubbleTeaOrder);
+}
```

Comment:
The part related to the AccountRoot is a nice plus but not a core requirement of the draft pattern.

Users can use the draft without having the user "remember" about the last edits.

I would suggest to move this part on the bottom and focus on the "core" of the pattern which is create a "Draft" object for the creation form.

Then explain that we can link the draft to the AccountRoot to have persistence of the unsaved edits.